### PR TITLE
test: cover factory with unit tests

### PR DESCRIPTION
**Pull request by warber**
#### What this PR does / Why we need it:
add basic unit tests to cover factory code

### CODE REVIEW

File: api/clients/factory_test.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+package clients
+
+import (
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/oauth2/clientcredentials"
+	"testing"
+)
+
+func TestClientCreation(t *testing.T) {
+	assert := assert.New(t)
```
Comment (UnseenWizzard) [inline api/clients/factory_test.go:24]: any deeper reason to go with an assert struct over the functional style of `assert.NoError(t, err)`?
I don't see any real benefit of this way tbh

File: api/clients/factory_test.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+			ClientID:     "client_id",
+			ClientSecret: "client_secret",
+			TokenURL:     "https://auth.example.com/token",
+		}).
+		WithUserAgent("MyUserAgent")
+
+	client, err := f.BucketClient()
+	assert.Error(err)
+	assert.Nil(client)
+	assert.Equal(ErrEnvironmentURLMissing, err)
```
Comment (UnseenWizzard) [inline api/clients/factory_test.go:57]: could use `assert.ErrorIs()` here

File: api/clients/factory_test.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+			ClientID:     "client_id",
+			ClientSecret: "client_secret",
+			TokenURL:     "https://auth.example.com/token",
+		}).
+		WithUserAgent("MyUserAgent")
+
+	client, err := f.BucketClient()
+	assert.Error(err)
+	assert.Nil(client)
+	assert.Contains(err.Error(), "failed to parse URL")
```
Comment (UnseenWizzard) [inline api/clients/factory_test.go:92]: could use ErrorContains here

`assert.ErrorContains(err, "failed to parse URL")`

File: api/clients/factory_test.go (hunk 1 above)
Comment (warber) [reply api/clients/factory_test.go:24]: not really it just got generated like this :P
i will change it

Comment (UnseenWizzard) [reply api/clients/factory_test.go]: 🤣 people write some weird code for AI to learn from...

File: api/clients/factory_test.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+	// Prepare a factory instance without OAuth credentials
+	f := Factory().
+		WithEnvironmentURL("https://example.com/api").
+		WithUserAgent("MyUserAgent")
+
+	client, err := f.BucketClient()
+	assert.Error(t, err)
+	assert.Nil(t, client)
+	assert.Equal(t, ErrOAuthCredentialsMissing, err)
```
Comment (UnseenWizzard) [inline api/clients/factory_test.go:69]: nit: could be ErrorIs as well

**Review by UnseenWizzard — APPROVED** [review]

File: api/clients/factory_test.go (hunk 4 above)
Comment (warber) [reply api/clients/factory_test.go]: sure :) pls re-approve

**Review by UnseenWizzard — APPROVED** [review]

**Pull request merged**